src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -0,0 +1,480 @@
+            {
+                index = 0;
+                hardSelect = false;
+            }
+
+            return new AsyncCompletionData.FilteredCompletionModel(
+                highlightedList, index, filters,
+                hardSelect ? AsyncCompletionData.UpdateSelectionHint.Selected : AsyncCompletionData.UpdateSelectionHint.SoftSelected,
+                centerSelection: true, uniqueItem: null);
+        }
+
+        private AsyncCompletionData.FilteredCompletionModel HandleAllItemsFilteredOut(
+            AsyncCompletionData.CompletionTriggerReason triggerReason,
+            ImmutableArray<AsyncCompletionData.CompletionFilterWithState> filters,
+            bool hasActiveFilters,
+            CompletionRules completionRules)
+        {
+            AsyncCompletionData.UpdateSelectionHint selection;
+            if (triggerReason == AsyncCompletionData.CompletionTriggerReason.Insertion)
+            {
+                // If the user was just typing, and the list went to empty *and* this is a 
+                // language that wants to dismiss on empty, then just return a null model
+                // to stop the completion session.
+                if (completionRules.DismissIfEmpty)
+                {
+                    return null;
+                }
+
+                selection = AsyncCompletionData.UpdateSelectionHint.NoChange;
+            }
+            else
+            {
+                // If the user has turned on some filtering states, and we filtered down to
+                // nothing, then we do want the UI to show that to them.  That way the user
+                // can turn off filters they don't want and get the right set of items.
+
+                // If we are going to filter everything out, then just preserve the existing
+                // model (and all the previously filtered items), but switch over to soft
+                // selection.
+                selection = hasActiveFilters ? AsyncCompletionData.UpdateSelectionHint.NoChange : AsyncCompletionData.UpdateSelectionHint.SoftSelected;
+            }
+
+            return new AsyncCompletionData.FilteredCompletionModel(
+                ImmutableArray<AsyncCompletionData.CompletionItemWithHighlight>.Empty, selectedItemIndex: 0,
+                filters, selection, centerSelection: true, uniqueItem: default);
+        }
+
+        private IEnumerable<AsyncCompletionData.CompletionItemWithHighlight> GetHighlightedList(
+            IEnumerable<ExtendedFilterResult> filterResults, 
+            string filterText,
+            bool highlightMatchingPortions)
+        {
+            var highlightedList = new List<AsyncCompletionData.CompletionItemWithHighlight>();
+            foreach (var item in filterResults)
+            {
+                var highlightedSpans = highlightMatchingPortions
+                                        ? _completionHelper.GetHighlightedSpans(item.VSCompletionItem.FilterText, filterText, CultureInfo.CurrentCulture)
```

Comment:
We're applying highlights to the DisplayText rather than FilterText. 
If the match is within FilterText but not DisplayText, then highlight will be wrong.
If the match is within FilterText and outside of the range of DisplayText, then VS will crash


I'm wrapping the UI code to prevent the crash, but the CompletionItemManager must return highlight against the DisplayText.